File: model/src/main/scala/ru/mipt/acsl/decode/model/domain/impl/types/ParameterImpl.scala
package ru.mipt.acsl.decode.model.domain.impl.types

import ru.mipt.acsl.decode.model.domain.LocalizedString
import ru.mipt.acsl.decode.model.domain.proxy.MaybeProxy
import ru.mipt.acsl.decode.model.domain.naming.ElementName
import ru.mipt.acsl.decode.model.domain.registry.DecodeUnit
import ru.mipt.acsl.decode.model.domain.types.DecodeType

/**
  * @author Artem Shein
  */
private[domain] class ParameterImpl(name: ElementName, info: LocalizedString,
                            val paramTypeProxy: MaybeProxy[DecodeType],
                            val unitProxy: Option[MaybeProxy[DecodeUnit]])
  extends AbstractNameInfoAware(name, info) with Parameter

File: model/src/main/java/ru/mipt/acsl/decode/model/ParameterImpl.java
package ru.mipt.acsl.decode.model;

import ru.mipt.acsl.decode.model.types.Alias;
import ru.mipt.acsl.decode.model.types.TypeMeasure;

/**
 * Created by metadeus on 06.06.16.
 */
public class ParameterImpl implements Parameter {

    private final Alias.MessageOrCommandParameter alias;
    private final TypeMeasure typeMeasure;

    ParameterImpl(Alias.MessageOrCommandParameter alias, TypeMeasure typeMeasure) {
        this.alias = alias;
        this.typeMeasure = typeMeasure;
    }

    @Override
    public Alias.MessageOrCommandParameter alias() {
        return alias;
    }

    @Override
    public TypeMeasure typeMeasure() {
        return typeMeasure;
    }
}
